storage/store/collections.go

```diff
@@ -0,0 +1,173 @@
+package store
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/operation"
+)
+
+type Collections struct {
+	db           storage.DB
+	transactions *Transactions
+	indexingByTx sync.Mutex
+}
+
+func NewCollections(db storage.DB, transactions *Transactions) *Collections {
+	c := &Collections{
```

Comment:
what do you think about adding a cache? collections are commonly looked up on access nodes. totally fine to do later

Reply:
Yeah, maybe add later.